Sources/TuistLoader/SwiftPackageManager/PackageInfo+graphPlatform.swift

```diff
@@ -4,23 +4,6 @@ import TuistSupport
 import XcodeGraph
 
 extension PackageInfo.Platform {
-    func graphPlatform() throws -> XcodeGraph.Platform {
-        switch platformName.lowercased() {
-        case "ios", "maccatalyst":
```

Comment:
This distinction is necessary.  SPM has a distinct platform defined for [maccatalyst](https://github.com/apple/swift-package-manager/blob/3b9dd49b637bb50aa4bbeada4d2deddef2601ae6/Sources/PackageModel/Platform.swift#L41) but in `XcodeGraph.Platform` we dont have `maccatalyst` as a distinct platform.  I believe that we'll miss any conditions that are `.when(. macCatalyst)`

Reply:
This is a failing test we can add to `SettingsMapperTests` that captures what should happen when a `maccatalyst` condition is used: 
```
    func test_set_maccatalyst() throws {
        let settings: [PackageInfo.Target.TargetBuildSettingDescription.Setting] = [
            .init(tool: .swift, name: .define, condition: nil, value: ["Define1"]),
            .init(
                tool: .swift,
                name: .define,
                condition: PackageInfo.PackageConditionDescription(platformNames: ["maccatalyst"], config: nil),
                value: ["Define2"]
            ),
        ]

        let mapper = SettingsMapper(
            headerSearchPaths: [],
            mainRelativePath: try RelativePath(validating: "path"),
            settings: settings
        )

        let allPlatformSettings = try mapper.settingsDictionary()

        XCTAssertEqual(
            allPlatformSettings["SWIFT_ACTIVE_COMPILATION_CONDITIONS"],
            .string("$(inherited) SWIFT_PACKAGE Define1")
        )

        let iosPlatformSettings = try mapper.settingsDictionary(for: .iOS)

        XCTAssertEqual(
            iosPlatformSettings["SWIFT_ACTIVE_COMPILATION_CONDITIONS"],
            .string("$(inherited) SWIFT_PACKAGE Define1 Define2")
        )

        let combinedSettings = try mapper.mapSettings()

        XCTAssertEqual(
            combinedSettings["SWIFT_ACTIVE_COMPILATION_CONDITIONS[sdk=iphoneos*]"],
            .string("$(inherited) SWIFT_PACKAGE Define1 Define2")
        )

        XCTAssertEqual(
            combinedSettings["SWIFT_ACTIVE_COMPILATION_CONDITIONS[sdk=iphonesimulator*]"],
            .string("$(inherited) SWIFT_PACKAGE Define1 Define2")
        )

        XCTAssertEqual(
            combinedSettings["SWIFT_ACTIVE_COMPILATION_CONDITIONS"],
            .string("$(inherited) SWIFT_PACKAGE Define1")
        )
    }
```

Reply:
Thanks a lot for catching this, you are right!

I went a bit different direction and kept this deleted. You can find the fix [here](https://github.com/tuist/tuist/pull/6386/commits/6d9ef05b324542026af4d5077c916272081551e4). The test now passes 🙌